packages/browser-wallet/src/popup/shared/IdCard/IdCard.tsx

```diff
@@ -0,0 +1,106 @@
+import clsx from 'clsx';
+import React, { useState } from 'react';
+
+import VerfifiedIcon from '@assets/svg/verified-stamp.svg';
+import RejectedIcon from '@assets/svg/rejected-stamp.svg';
+import CheckIcon from '@assets/svg/checkmark-blue.svg';
+import EditIcon from '@assets/svg/edit.svg';
+import ConcordiumIcon from '@assets/svg/concordium-small.svg';
+import { useTranslation } from 'react-i18next';
+import Form from '../Form';
+import Submit from '../Form/Submit';
+import Button from '../Button';
+import FormInlineInput from '../Form/InlineInput';
+
+const IDENTITY_NAME_MAX_LENGTH = 20;
+
+type EditNameForm = {
+    name: string;
+};
+
+type EditNameProps = {
+    name: string;
+    onChange(name: string): void;
+};
+
+function EditableName({ name, onChange }: EditNameProps) {
+    const [isEditing, setIsEditing] = useState(false);
+
+    const handleSubmit = (f: EditNameForm) => {
+        onChange(f.name);
+        setIsEditing(false);
+    };
+
+    return (
+        <Form<EditNameForm> onSubmit={handleSubmit} className="id-card__name">
+            {(f) => (
+                <>
+                    {isEditing ? (
+                        <FormInlineInput
+                            name="name"
+                            control={f.control}
+                            className="id-card__name-field"
+                            defaultValue={name}
+                            fallbackValue={name}
+                            autoFocus
+                            rules={{
+                                required: true,
+                                maxLength: IDENTITY_NAME_MAX_LENGTH,
+                            }}
+                        />
+                    ) : (
+                        name
+                    )}
+                    {isEditing ? (
+                        <Submit className="id-card__name-edit" clear>
+                            <CheckIcon />
+                        </Submit>
+                    ) : (
+                        <Button className="id-card__name-edit" clear onClick={() => setIsEditing(true)}>
+                            <EditIcon />
+                        </Button>
+                    )}
+                </>
+            )}
+        </Form>
+    );
+}
+
+type Props = {
+    name: string;
+    status: 'pending' | 'approved' | 'rejected';
```

Comment:
Two things here:
1. The status of an identity should be defined centrally as an enum, as we will be using that in multiple places in the application.
2. In the desktop wallet we use `confirmed` over `approved` and we should stay consistent to avoid confusing ourselves with any differences between the two terms. (Event though I like the approved terminology better :smile:).

Context: The identity provider actually uses the values: 'pending', 'error' and 'done', but it seems that we went with a translation of these values to something that fits the wallet side better. I also think that `done` would be a bit weird to have as the status internally.

Reply:
1. I agree, however I thought it made better sense to wait until the identity object would be used, and then adapt the `IdCard` component to use the enum.
2. Alright. I'll rename it to confirmed.